README.md

```diff
@@ -81,30 +81,64 @@ This is only a list of approved features that will be included in Tirith over th
 pip install git+https://github.com/StackGuardian/tirith.git
 ```
 
+# Development Container Setup
+
+This guide will help you set up and run a development container.
+
+## Prerequisites
+
+- **Docker**: Ensure you have Docker installed on your machine. Download and install it from [Docker's official site](https://www.docker.com/products/docker-desktop).
+- **VS Code**: You should have Visual Studio Code installed.
+
+**Install the [Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) in VS Code.**
+
+## Running the Dev Container
+
+- Clone the repository to your local machine:
+
+```bash
+   git clone <repository-url>
+   cd <repository-folder>
+```
+
+- Open the project folder in Visual Studio Code
+
+- Once inside VS Code, open the Command Palette `(Ctrl+Shift+P or Cmd+Shift+P on macOS)` and search for **Dev Containers: Rebuild and Reopen in Container**. Select this option.
+
+- VS Code will build the dev container based on the devcontainer.json file or Docker configuration provided in the project. This may take a few minutes.
+
+- Once the container is up, you will have a fully configured development environment running inside Docker.
+
 ### For developers
+
```

Comment:
There are a lot of formatting changes here, could you please not include it in this commit. We can maybe have a separate issue regarding this (adding a doc linter, but need discussion first)